src/main/java/com/myaudiolibrary/web/controller/ArtistsController.java

```diff
@@ -0,0 +1,240 @@
+package com.myaudiolibrary.web.controller;
+
+
+import com.myaudiolibrary.web.model.Album;
+import com.myaudiolibrary.web.model.Artist;
+import com.myaudiolibrary.web.repository.ArtistRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Sort;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.ModelMap;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.servlet.view.RedirectView;
+
+import javax.persistence.EntityExistsException;
+import javax.persistence.EntityNotFoundException;
+import java.awt.print.Pageable;
+import java.util.Optional;
+
+@Controller
+@RequestMapping("/artists")
+public class ArtistsController {
+
+    @Autowired
+    ArtistRepository artistRepository;
+
+    /**
+     * Appeler lorque l'on souhaite afficher un artiste
+     * @param id De l'artiste
+     * @param model artiste retourner a la vue (detailArtist.html)
+     * @return
+     */
+    @GetMapping(value = "/{id}")
+    public String getArtistById(@PathVariable("id") Integer id, final ModelMap model){
+        // catch erreur de conversion ( si artist/A ... )
+
+        // erreur 404 not found
+        if(!artistRepository.existsById(id)){
+            throw new EntityNotFoundException("L'employé d'identifiant " + id + " n'a pas été trouvé !");
+        }
+
+        Optional<Artist> artistOptional = artistRepository.findById(id);
```

Comment:
Utiliser plutôt à fond l'optional pour éviter le double appel BDD
```suggestion
        Optional<Artist> artistOptional = artistRepository.findById(id);
        if(!artistOptional.isPresent()){
            throw new EntityNotFoundException("L'employé d'identifiant " + id + " n'a pas été trouvé !");
       }
```